code/common/timer.q

```diff
@@ -74,24 +73,28 @@ runandreschedule:{
 		update lastrun:start,nextrun:n from `.timer.timer where id=x`id;
 		[if[debug;.lg.o[`timer;"setting timer ID ",(string x`id)," to inactive as next schedule time is outside of scheduled period"]];
 		 update lastrun:start,active:0b from `.timer.timer where id=x`id]];
+	.timer.nextruntime:exec min[nextrun] from .timer.timer;
 	}
 
 //Set .z.ts
 if[.timer.enabled;
  $[@[{value x;1b};`.z.ts;0b];
    .z.ts:{[x;y] .timer.run now:.proc.cp[]; x@y}[.z.ts];
-   .z.ts:{.timer.run[.proc.cp[]]}];
+   .z.ts:{if[.proc.cp[]>.timer.nextruntime;.timer.run[.proc.cp[]]]}];
 
  // Set the timer to 200ms if not set already
  if[not system"t"; system"t 200"]];
-	
+if[.proc.lowpowermode;
```

Comment:
could you add the error trap around this e.g. 

@[value;`.proc.lowerpowermode;0b]